src/metabase/domain_entities/card.cljc

```diff
@@ -0,0 +1,260 @@
+    [:collection {:optional true} :any]
+    [:collection_id [:maybe number?]]
+    [:collection_position [:maybe number?]]
+    [:collection_preview boolean?]
+    [:created_at [:maybe string?]] ;; TODO: Date regex?
+    [:creator {:optional true}
+     [:map
+      [:id number?]
+      [:common_name string?]
+      [:date_joined string?] ;; TODO: Date regex
+      [:email string?]
+      [:first_name  [:maybe string?]]
+      [:is_qbnewb boolean?]
+      [:is_superuser boolean?]
+      [:last_login string?] ;; TODO: Date regex
+      [:last_name   [:maybe string?]]]]
+    [:creator_id {:optional true} number?]
+    [:creation-type {:optional true :js/prop "creationType"} string?]
+    [:dashboard_count {:optional true} number?]
+    [:dashboard-id {:optional true :js/prop "dashboardId"} number?]
+    [:dashcard-id  {:optional true :js/prop "dashcardId"}  number?]
+    [:database_id {:optional true} number?]
+    [:dataset {:optional true} boolean?]
+    [:dataset_query ::lib.schema/query] ;; NOTE: This is not converted using Malli; see CardNoQuery.
+    [:description [:maybe string?]]
+    ;; TODO: Display is really an enum but I don't know all its values.
+    ;; Known values: table, scalar, gauge, map, area, bar, line. There are more missing for sure.
+    [:display string?]
+    [:display-is-locked {:optional true :js/prop "displayIsLocked"} boolean?]
+    [:embedding_params {:optional true} :any]
+    [:enable_embedding {:optional true} boolean?]
+    [:entity_id {:optional true} string?]
+    [:id {:optional true} [:or number? string?]]
+    [:last-edit-info {:optional true :js/prop "last-edit-info"} :any]
+    [:last-query-start {:optional true} :any]
+    [:made_public_by_id {:optional true} [:maybe number?]]
+    [:moderation_reviews {:optional true} [:vector :any]]
+    [:name {:optional true} string?]
+    [:original_card_id {:optional true} number?]
+    [:original_card_name {:optional true :js/prop "originalCardName"} string?]
+    [:parameter_mappings {:optional true} [:vector :any]]
+    [:parameter_usage_count {:optional true} number?]
+    [:parameters {:optional true} [:sequential Parameter]]
+    [:persisted {:optional true} boolean?]
+    [:public_uuid {:optional true} [:maybe string?]]
+    [:query_type {:optional true} string?] ;; TODO: Probably an enum
+    [:result_metadata {:optional true} :any]
+    [:series_index {:optional true :js/prop "_seriesIndex"} :any]
+    [:series_key {:optional true :js/prop "_seriesKey"} :any]
+    [:table_id {:optional true} [:maybe number?]]
+    [:updated_at {:optional true} string?] ;; TODO: Date regex
+    [:visualization_settings VisualizationSettings]]])
+
+#?(:cljs
+   (def ^:private CardNoQuery
+     "Modified schema for Card, with the `:dataset_query` key set to :any.
+     Queries are special and we want Malli to disregard them."
+     (-> Card
+         mc/schema
+         mc/children
+         first
+         (mut/assoc-in [:dataset_query] :any))))
```

Comment:
Could you have `CardNoQuery` be the base version and then have `Card` just be that plus `mu/merge` the schema that includes `:dataset_query`? Seems like it would mean you'd need to dig into Malli internals a little less